_posts/2015-06-10-d8-theming-basics.markdown

```diff
@@ -0,0 +1,376 @@
+
+- {% raw %}`{{ These }}`{% endraw %} are for printing content, either explicitly or via functions
+- {% raw %}`{% These %}`{% endraw %} are for executing statements
+- {% raw %}`{# These #}`{% endraw %} are for comments
+
+#### Printing variables and regions
+
+In Drupal 7 we render content like so:
+
+{% highlight php %}
+<?php print render($page['sidebar']); ?>
+{% endhighlight %}
+
+Printing variables using Twig in D8 is as easy as including them in the double curly brace delimiter.
+
+{% highlight liquid %}
+// In page--front.html.twig
+// Print the sidebar region.
+{% raw %}
+  {{ page.sidebar }}
+{% endraw %}
+{% endhighlight %}
+
+...unless there are special characters in the variable name. If that's the case and you see an error when using the syntax above, you can use Twig's subscript syntax, which should look pretty familiar to Drupalers:
+
+{% highlight liquid %}
+// In page--front.html.twig
+// Print the page type.
+{% raw %}
+  {{ page['#type'] }}
+{% endraw %}
+{% endhighlight %}
+
+This will be more useful when debugging. The Drupal core base classes include lists of available variables and regions in the DocBlock of their template files, or you can print variables to the page via Twig's debug mode (more on that below) to see what's available to you.
+
+#### Filters and functions
+
+Twig comes with many built-in [filters](http://twig.sensiolabs.org/doc/filters/index.html) that variables are passed to via the pipe character. These filters do many of the things that PHP functions would have in previous Drupal versions. One example is the date filter:
+
+{% highlight liquid %}
+// Format the post date.
+{% raw %}
+  {{ post.published|date("Y-m-d") }}
+{% endraw %}
+{% endhighlight %}
+
+There are also [Drupal-specific Twig filters](https://www.drupal.org/node/2357633), such as `t` which runs the string through the `t()` function.
+{% highlight liquid %}
+// Run an ARIA label through t()
+{% raw %}
+  <nav class="tabs" role="navigation" aria-label="{{ 'Tabs'|t }}">
+{% endraw %}
+{% endhighlight %}
+
+By the way, ARIA labels are new in Drupal 8 too!
+
+In addition to filters, Twig provides a range of [functions](http://twig.sensiolabs.org/doc/functions/index.html) that are also used within the double curly brace delimiters.
+
+#### Tags
+
+Control flow and other [tags](http://twig.sensiolabs.org/doc/tags/set.html) are also supported in Twig. One of my favorite things about templating languages is how easy it is to execute `if` statements and `for` loops. Savas uses Jekyll for our company website and the Liquid templating language makes it easy to loop through a list of data points, blog posts, projects, etc. and print them to a page rather than writing out all of the HTML. In Drupal, we'll use the `if` statement quite often.
```

Comment:
"Savas uses Jekyll for our company website" might be good to link to our github repo here